dagger/blockexchange/engine.nim

```diff
@@ -112,62 +143,153 @@ proc stop*(b: BlockExcEngine) {.async.} =
 
   trace "NetworkStore stopped"
 
+proc toDiscoveryId*(cid: Cid): NodeId =
+  ## To discovery id
+  readUintBE[256](keccak256.digest(cid.data.buffer).data)
```

Comment:
It would be much better, if we can avoid the double hashing here. The `Cid` already contains a hash - we're using `sha-256`...

Reply:
Yeah, would be ideal, but I think we need to modify libp2p to expose the hash without multihash headers
I can do it in a follow up PR, not urgent, I think

Reply:
It's already exposed, albeit a bit chunkily, this is how you access it - `mh.data.buffer[mh.dpos..(mh.dpos + mh.size - 1)]`.

Reply:
> Yeah, would be ideal, but I think we need to modify libp2p to expose the hash without multihash headers I can do it in a follow up PR, not urgent, I think

I would agree that we probably ok with the way things are for now, on a second thought, this might be the desired way of doing it either way...